### ISSUE TITLE
Hotpep issue

### ISSUE THREAD

**Issue by mpunzalan** [linnabrown/run_dbcan#4]
Hi! 

I ran run_dbcan.py and all the other tools are running properly except for Hotpep:

$python run_dbcan.py ./test/gh50_clade.fasta protein --out_dir ./test
***************************1. DIAMOND start*************************************************

***************************2. HMMER start*************************************************

diamond v0.9.21.122 | by Benjamin Buchfink <buchfink@gmail.com>
Licensed under the GNU AGPL <https://www.gnu.org/licenses/agpl.txt>
Check http://github.com/bbuchfink/diamond for updates.

#CPU threads: 5
Scoring parameters: (Matrix=BLOSUM62 Lambda=0.267 K=0.041 Penalties=11/1)
Temporary directory: ./test
Opening the database... 
***************************3. HotPep start***************************************************

Screening  for
CE
diamond v0.9.21.122 | by Benjamin Buchfink <buchfink@gmail.com>
Licensed under the GNU AGPL <https://www.gnu.org/licenses/agpl.txt>
Check http://github.com/bbuchfink/diamond for updates.

#CPU threads: 5
Scoring parameters: (Matrix=BLOSUM62 Lambda=0.267 K=0.041 Penalties=11/1)
Temporary directory: ./test
Opening the database...  [0.000434s]
#Target sequences to report alignments for: 1
Opening the input file...  [0.000153s]
Opening the output file...  [0.000122s]
Loading query sequences...  [0.000288s]
Masking queries...  [0.001335s]
Building query seed set...  [0.002955s]
Algorithm: Query-indexed
Building query histograms...  [0.000603s]
Allocating buffers...  [1.8e-05s]
Loading reference sequences... Traceback (most recent call last):
  File "parallel_group_many_proteins_many_patterns_noDNA.py", line 24, in <module>
    hit_cut_off = int(sys.argv[5])
ValueError: invalid literal for int() with base 10: '2.6'
GH
Traceback (most recent call last):
  File "parallel_group_many_proteins_many_patterns_noDNA.py", line 24, in <module>
    hit_cut_off = int(sys.argv[5])
ValueError: invalid literal for int() with base 10: '2.6'
AA
Traceback (most recent call last):
  File "parallel_group_many_proteins_many_patterns_noDNA.py", line 24, in <module>
    hit_cut_off = int(sys.argv[5])
ValueError: invalid literal for int() with base 10: '2.6'
PL
Traceback (most recent call last):
  File "parallel_group_many_proteins_many_patterns_noDNA.py", line 24, in <module>
    hit_cut_off = int(sys.argv[5])
ValueError: invalid literal for int() with base 10: '2.6'
GT
Traceback (most recent call last):
  File "parallel_group_many_proteins_many_patterns_noDNA.py", line 24, in <module>
    hit_cut_off = int(sys.argv[5])
ValueError: invalid literal for int() with base 10: '2.6'
CBM
Traceback (most recent call last):
  File "parallel_group_many_proteins_many_patterns_noDNA.py", line 24, in <module>
    hit_cut_off = int(sys.argv[5])
ValueError: invalid literal for int() with base 10: '2.6'

Screened

for proteins of the types
CE, GH, AA, PL, GT, CBM
 [2.15639s]
Building reference histograms...  [0.650409s]
Allocating buffers...  [1.7e-05s]
Initializing temporary storage...  [0.0447s]
Processing query chunk 0, reference chunk 0, shape 0, index chunk 0.
Building reference index...  [0.558347s]
Building query index...  [0.00069s]
Building seed filter...  [0.002714s]
Searching alignments...  [0.055795s]
Deallocating buffers...  [0.001044s]
Computing alignments...  [0.022396s]
Deallocating reference...  [0.055838s]
Loading reference sequences...  [1.5e-05s]
Deallocating buffers...  [0.000323s]
Deallocating queries...  [0.000701s]
Loading query sequences...  [9e-06s]
Closing the input file...  [1.1e-05s]
Closing the output file...  [0.000742s]
Closing the database file...  [7e-06s]
Deallocating taxonomy...  [2e-06s]
Total time = 3.56039s
Reported 7 pairwise alignments, 7 HSPs.
7 queries aligned.
***************************1. DIAMOND end***************************************************
***************************2. HMMER end***************************************************
***************************3. hotPep end***************************************************
Waiting on signalP
SignalP complete
Preparing overview table from hmmer, hotpep and diamond output...
overview table complete. Saved as ./test/overview.txt

I got a blank Hotpep.out file with just the headers. I'm certain that my sequences have Hotpep results in them as I checked them in the online annotation site of dbcan meta.
I ran the 1st version of run_dbcan.py before and I have never encountered these Value Errors during the HotPep run. 

Thank you!

**Comment by linnabrown** [comment]
@HaidYi is writing a docker version. Coming soon.

**Comment by linnabrown** [comment]
`python run_dbcan.py ./test/gh50_clade.fasta protein --out_dir ./test`

Your directory name is incorrect.
you should try this one:

`python run_dbcan.py test/gh50_clade.fasta protein --out_dir test`

**Comment by linnabrown** [comment]
`1. Make sure docker is installed on your computer successfully.
2. docker pull haidyi/run_dbcan:latest
3. docker run --name <preferred_name> -v <host-path>:<container-path> -it haidyi/run_dbcan:latest python run_dbcan.py <input_file> [params] --out_dir <output_dir>`

**Comment by linnabrown** [comment]
```
1. Make sure docker is installed on your computer successfully.
2. docker pull haidyi/run_dbcan:latest
3. docker run --name <preferred_name> -v <host-path>:<container-path> -it haidyi/run_dbcan:latest python run_dbcan.py <input_file> [params] --out_dir <output_dir>
```

**Comment by HaidYi** [comment]
@mpunzalan  if you meet some permission problems in docker, use the command
```
sudo chown user:user <out_dir>
```
to change the owner of out_dir.

**Comment by BenedicteColnet** [comment]
Hello ! 

As @mpunzalan I have the same error for Hotpep output with the error message :

``` ValueError: invalid literal for int() with base 10: '2.6' ```

and at the end no output created. (no blank Hotpep.out file)

I had this problem when using my own Docker image, and I thought that using yours would solve the problem. But I still have the same error message. 

For your information, when running run_dbcan with the EscherichiaColi example provided I don't have error message.

Thank you by advance for your help !

**Comment by BenedicteColnet** [comment]
I was really curious about this error, therefore I investigated Hotpep code and dbCAN code. It seems that the error comes from a missing argument when you have a path `./test/gh50_clade.fasta` instead of a single fasta file `gh50_clade.fasta` because of an incomplete parsing.

I fixed this issue changing the code in ` run_dbcan.py` line 173 :

```
  inter = input.split('/')[-1]
  directory = inter.split('.')[0]
```

instead of 

```
directory = input.split('.')[0]
```

That would be great if someone could confirm I did well doing this.

Thank you by advance.
